### PR TITLE
Combine Sync & Data Indexers

### LINKED ISSUE #521: Combine data and sync indexers & ditch the separate sync indexer
We are updating the index table multiple times and it causes the indexers to run multiple times. Let's ditch the sync indexer for good and simply sync the products along with the data indexer.

- Remove subscriptions, mview and plugins for Sync indexer
- Call product sync after each batch here https://github.com/Nosto/nosto-magento2/blob/feature/product-caching-index/Model/Service/Index.php#L193

### PR DESCRIPTION
**Pull request by supercid**
## Description
Combine sync and data indexers into a single indexer, and remove the sync observer that was used on update on save. This is already being done in #514 

## Related Issue
Fixes #521 

## Motivation and Context
The sync process could have been combined for simplicity. There's also no data being indexed, only being transferred via API.

## How Has This Been Tested?
Run `bin/magento indexer:reset nosto_index_product_data_sync; bin/magento indexer:reindex nosto_index_product_data_sync`
and observe the logs and product payload sent.

## Documentation:
N/A

## Checklist:
- [X] My code follows the code style of this project.
- [X] I have updated the documentation accordingly.
- [X] All new and existing tests passed.
- [X] I have assigned the correct milestone or created one if non-existent.
- [X] I have correctly labeled this pull request.
- [X] I have linked the corresponding issue in this description.
- [X] I have updated the corresponding Jira ticket.
- [X] I have requested a review from at least 2 reviewers
- [X] I have checked the base branch of this pull request
- [X] I have checked my code for any possible security vulnerabilities

### CODE REVIEW

File: Model/Indexer/Data.php [hunk 1]
```diff
@@ -129,4 +148,33 @@ private function getCollection(array $ids = [])
...
+    /**
+     * Returns a collection Nosto product index items that are out of sync and not deleted.
+     * If $ids attribute is present the collection will be limited to matching the ids and the
+     * condition mentioned above only.
+     *
+     * @param Store|null $store
+     * @param array $ids
+     * @return IndexCollection
+     */
+    private function getOutOfSyncCollection(Store $store = null, array $ids = [])
```
Comment (hannunosto) [inline Model/Indexer/Data.php]: This should return both out of sync and dirty ones

File: Model/Indexer/Data.php [hunk 2]
```diff
@@ -74,8 +82,13 @@ public function __construct(
      */
     public function executeFull()
     {
-        $indexCollection = $this->getCollection();
-        $this->nostoServiceIndex->handleDirtyProducts($indexCollection);
+        $dirtyCollection = $this->getDirtyCollection();
```
Comment (hannunosto) [inline Model/Indexer/Data.php]: Let's move to sync logic to the index service

File: Model/Indexer/Data.php [hunk 3]
```diff
@@ -42,30 +42,38 @@
...
+use Nosto\Tagging\Helper\Account as NostoHelperAccount;
+use Magento\Store\Model\Store;
 
 /**
  * An indexer for Nosto product sync
  */
 class Data implements IndexerActionInterface, MviewActionInterface
 {
-    public const INDEXER_ID = 'nosto_product_index_data';
+    public const INDEXER_ID = 'nosto_index_product_data_sync';
```
Comment (mridang) [inline Model/Indexer/Data.php]: Will there be any issues with the upgrade?

Comment (supercid) [reply Model/Indexer/Data.php]: No.

**Pull request merged**